Variable selector

dmfalke opened the pull request:
This PR adds an `InputVariables` component, which renders a set of `VariableTreeDropdown` components. It also incorporates computation app metadata into the **Visualize** tab.

Depends on https://github.com/VEuPathDB/web-components/pull/100
Fixes #25 #41

dmfalke commented:
@bobular pinging you for a review :)

bobular commented on src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx:
```diff
@@ -444,17 +428,16 @@ export function histogramResponseToData(
-  overlayVariable?: StudyVariable, // TO DO: ?CategoricalVariable?
+  variable: Variable,
+  variableType: 'number' | 'date',
+  overlayVariable?: Variable,
```
Well this is certainly much cleaner - not passing variable and entity for both variables!

bobular reviewed (approved):
There's a lot here!

I've looked through all the diff and understand the main changes but haven't reviewed every line!

Functionally it's perfect except for the placeholder cartoon histogram being fleetingly very large in grid view (when it shows visualisations you've already made).  I will see if I can put in a suggestion or commit to fix this during my next meeting.

What's the plan for using the computation's name (e.g. `pass`) in the visualisation's get-data call? I notice the `/pass/` is still hardcoded in data-api.ts   Will that be another ticket?

bobular commented on src/lib/workspace/ComputationRoute.tsx:
```diff
@@ -11,6 +12,14 @@ export interface Props {
+  const computationAppOverview = computations.value?.apps.find(
+    (app) => app.name === 'pass'
+  );
+  if (computationAppOverview == null) return null;
```
I was also wondering if we need a `<Suspense>` for this bit?
Or are we assuming the getApps() call will always be fast?

dmfalke replied:
It should be pretty fast. I don't think this endpoint does any database lookups, but we should still handle possible slow responses. I'll add this.

dmfalke commented:
Ah, I guess I forgot about using the computation name for the data call. This isn't really urgent since we don't have other computations yet. It's probably a good idea to deal with it now.

bobular commented:
Another thing is that the OverlayVariable `clear` link clears both variables, not just the overlay one

![image](https://user-images.githubusercontent.com/308639/114869554-09144500-9def-11eb-8fac-778f7056fbd7.png)

dmfalke commented:
> Another thing is that the OverlayVariable `clear` link clears both variables, not just the overlay one

D'oh! Good catch. I will also address soon.

dmfalke commented:
@bobular I've addressed the issues you raised:

* Loading indicator for apps data (note that Suspense is not production ready for data fetching. I added a new component `PromiseResult` that can be used with the return value of the `usePromise` hook.
* Use computation `type` property when fetch visualization data.
* Fix the **clear** button in `InputVariables.tsx`

I am going to merge. Thanks!
